roles/awx/files/awx-deployment.yml

```diff
@@ -0,0 +1,133 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: awx-deployment
+parameters:
+- name: NAMESPACE
+- name: AWX_WEB_IMAGE
+- name: AWX_TASK_IMAGE
+- name: AWX_ADMIN_PASSWORD
+
+objects:
+
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: admin-credentials
+    namespace: ${NAMESPACE}
+  stringData:
+    username: admin
+    password: ${AWX_ADMIN_PASSWORD}
+
+- apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
+    name: awx
+    namespace: ${NAMESPACE}
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          name: awx-web-deploy
+          service: django
+      spec:
+        containers:
+          - name: awx-web
+            image: ${AWX_WEB_IMAGE}
+            ports:
+              - containerPort: 8052
+            volumeMounts:
+              - mountPath: /etc/tower
+                name: awx-application-config
+            env:
+              - name: DATABASE_PASSWORD
```

Comment:
Environment is not a secure place for secrets... can awx consume the password from a file on disk?

Reply:
It probably can — I can mount the secret as volume, and modify settings.py to read the files.

But can you explain the threat model?
k8s API returns such env vars with `valueFrom: secretKeyRef:`, it does not expose the effective value.
Are you thinking of with `oc exec` access into the pod — I think then it's game over anyway?
Does the kubelet leak them in some way?

Many of the builtin templates use such `secretKeyRef` env vars for keys, passwords, etc.:
https://github.com/openshift/openshift-ansible/search?q=secretKeyRef
including the postgresql-persistent template this role uses:
https://github.com/openshift/openshift-ansible/blob/5c1207cd2c63736e93168e77c30a6b0fbd05afc9/roles/openshift_examples/files/examples/v3.10/db-templates/postgresql-persistent-template.json#L123-L127

Reply:
A lot of the issues with secrets in env were covered way back in [the original design discussion](https://github.com/kubernetes/kubernetes/issues/4710), but I wouldn't be surprised if various enhancements have been made since then that I'm just unaware of to reduce the risks.

In this case I probably wouldn't go too far out of my way to introduce file support where none exists, but I would default to avoiding environment when possible (as it's generally an additional risk with no clear added value outside shim use cases).

Reply:
No can do, not with current awx images.
Turns out they look at POSTGRESQL_PASSWORD, AWX_ADMIN_USER, AWX_ADMIN_PASSWORD in shell scripts (that are baked into the image) before settings.py (that we control) comes into play.
Can't even attempt equivalent logic in settings.py later, because if we omit AWX_ADMIN_PASSWORD env var, we might later create a superuser with password specified via a file, but an insecure `admin`/`password` superuser will be created *as well*.

I'll start a discussion upstream in awx about various ways (including this) learnt from here to improve their k8s/openshift deployment security, and see if they want PRs for any of those.